feat(*): add to travis

arschles opened the pull request:
Fixes #12 

cc/ @mboersma @krancour

arschles committed:
feat(deploy.sh): add deploy script

arschles committed:
feat(Makefile): add test target

to work with a forthcoming travis file

arschles committed:
feat(.travis.yml): add travis file

arschles committed:
feat(README.md): add travis and go report card badges

arschles committed:
fix(.travis.yml): install glide before running tests

krancour commented on .travis.yml:
```diff
@@ -0,0 +1,24 @@
+install:
+  - GLIDE_HOME=/home/travis/.glide make bootstrap
+script:
+  - make bootstrap test
```
This will invoke the `bootstrap` target a second time, unnecessarily.

arschles replied:
ah yea, thanks

arschles replied:
I actually overlooked something here - I'm gonna use `quay.io/deis/go-dev` instead of installing glide here

krancour replied:
Even better!  I didn't want to force that on people, but :100: !

krancour replied:
And as I have just learned, you can change the language to `generic` then to start with no language stack.  This is largely undocumented, but @sgoings discovered it earlier today.

arschles replied:
:+1: changes coming after I finish testing all this locally

arschles committed:
fix(Makefile): use go-dev image for the go and glide tools

arschles committed:
fix(.travis.yml): use language: generic in travis file

because all tools, including the go toolchain, are encapsulated inside
the quay.io/deis/go-dev container

arschles commented:
@krancour changes made to use quay.io/deis/go-dev
